.NET/Microsoft.Recognizers.Text.DataTypes.DataDrivenTests/TestTimex.cs

```diff
@@ -17,138 +17,112 @@ public void DataTypes_Timex_FromDate()
+        public void DataTypes_Timex_Roundtrip_DateTime(string timex)
         {
-            Roundtrip("XXXX-WXX-3T04");
-            Roundtrip("2017-09-27T11:41:30");
+            Roundtrip(timex);
         }
 
         [TestMethod]
-        public void DataTypes_Timex_Roundtrip_DateRange()
+        [DataRow("2017")]
+        [DataRow("SU")]
+        [DataRow("2017-WI")]
+        [DataRow("2017-09")]
+        [DataRow("2017-W37")]
+        [DataRow("2017-W37-WE")]
+        [DataRow("XXXX-05")]
+        public void DataTypes_Timex_Roundtrip_DateRange(string timex)
         {
-            Roundtrip("2017");
-            Roundtrip("SU");
-            Roundtrip("2017-WI");
-            Roundtrip("2017-09");
-            Roundtrip("2017-W37");
-            Roundtrip("2017-W37-WE");
-            Roundtrip("XXXX-05");
+            Roundtrip(timex);
         }
 
         [TestMethod]
-        public void DataTypes_Timex_Roundtrip_DateRange_Start_End_Duration()
+        [DataRow("(XXXX-WXX-3,XXXX-WXX-6,P3D)")]
+        [DataRow("(XXXX-01-01,XXXX-08-05,P216D)")]
+        [DataRow("(2017-01-01,2017-08-05,P216D)")]
+        [DataRow("(2016-01-01,2016-08-05,P217D)")]
+        public void DataTypes_Timex_Roundtrip_DateRange_Start_End_Duration(string timex)
         {
-            Roundtrip("(XXXX-WXX-3,XXXX-WXX-6,P3D)");
-            Roundtrip("(XXXX-01-01,XXXX-08-05,P216D)");
-            Roundtrip("(2017-01-01,2017-08-05,P216D)");
-            Roundtrip("(2016-01-01,2016-08-05,P217D)");
+            Roundtrip(timex);
         }
 
         [TestMethod]
-        public void DataTypes_Timex_Roundtrip_TimeRange()
+        [DataRow("TEV")]
+        public void DataTypes_Timex_Roundtrip_TimeRange(string timex)
         {
-            Roundtrip("TEV");
+            Roundtrip(timex);
         }
 
         [TestMethod]
-        public void DataTypes_Timex_Roundtrip_TimeRange_Start_End_Duration()
+        [DataRow("(T16,T19,PT3H)")]
+        public void DataTypes_Timex_Roundtrip_TimeRange_Start_End_Duration(string timex)
         {
-            Roundtrip("(T16,T19,PT3H)");
+            Roundtrip(timex);
         }
 
         [TestMethod]
-        public void DataTypes_Timex_Roundtrip_DateTimeRange()
+        [DataRow("2017-09-27TEV")]
+        public void DataTypes_Timex_Roundtrip_DateTimeRange(string timex)
         {
-            Roundtrip("2017-09-27TEV");
+            Roundtrip(timex);
         }
 
         [TestMethod]
-        public void DataTypes_Timex_Roundtrip_DateTimeRange_Start_End_Duration()
+        [DataRow("(2017-09-08T21:19:29,2017-09-08T21:24:29,PT5M)")]
+        [DataRow("(XXXX-WXX-3T16,XXXX-WXX-6T15,PT71H)")]
+        public void DataTypes_Timex_Roundtrip_DateTimeRange_Start_End_Duration(string timex)
         {
-            Roundtrip("(2017-09-08T21:19:29,2017-09-08T21:24:29,PT5M)");
-            Roundtrip("(XXXX-WXX-3T16,XXXX-WXX-6T15,PT71H)");
-        }
-
-        [TestMethod]
-        public void DataTypes_Timex_ToString()
```

Comment:
What happened to the cases from this one onwards?

Reply:
I moved them to `TestTimexToString` because all the tests before this are pure roundtrip tests and it was getting hard to track what tests tested what codepaths.